flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/sql/SqlSessionTableFunction.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.functions.sql;
+
+import org.apache.flink.shaded.guava30.com.google.common.collect.ImmutableList;
+
+import org.apache.calcite.sql.SqlCallBinding;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlOperator;
+
+/**
+ * SqlSessionTableFunction implements an operator for session.
+ *
+ * <p>It allows three parameters:
+ *
+ * <ol>
+ *   <li>a table
+ *   <li>a descriptor to provide a time attribute column name from the input table
+ *   <li>an interval parameter to specify an inactive activity gap to break sessions
+ * </ol>
+ */
+public class SqlSessionTableFunction extends SqlWindowTableFunction {
+
+    public SqlSessionTableFunction() {
+        super(SqlKind.SESSION.name(), new OperandMetadataImpl());
+    }
+
+    /** Operand type checker for SESSION. */
+    private static class OperandMetadataImpl extends AbstractOperandMetadata {
+        OperandMetadataImpl() {
+            super(ImmutableList.of(PARAM_DATA, PARAM_TIMECOL, PARAM_STEP, PARAM_SIZE), 3);
```

Comment:
What does the 4th parameter do? It seems that a session window only needs 3 parameters.

Reply:
good catch.